### PR TITLE
Create Cartoon-type BC

### PR DESCRIPTION
**Pull request by michaeldmurphy1**
## Proposed changes

Elements neighboring the $y$-axis use ZernikeB1 basis when doing cartoon, which does not need a boundary condition. However, when using subcell we do need to give ghost data. This PR makes a BC type that will be skipped for the DG time derivatives but properly plugs in FG ghost data. 

Systems without subcell can put the default `Cartoon` BC in their factory list and be fine (currently just ScalarWave) while systems using subcell will have to create a system-specific BC (currently just Valencia and GhValencia)

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.
- [ ] If a coding agent is used, have one of
      "Co-Authored-By: Claude Sonnet 4.6 <noreply@anthropic.com>",
      "Co-Authored-by: Codex <noreply@openai.com>", or
      "Co-Authored-By: GitHub Copilot CLI <noreply@microsoft.com>"
      as the last line of the commit, depending on the agent.

### CODE REVIEW

File: src/Domain/BoundaryConditions/Cartoon.hpp [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+ * set-up to use the cartoon method. It should not be used as an external
+ * boundary.
+ */
+template <typename SystemBoundaryConditionBaseClass>
+struct Cartoon final : public SystemBoundaryConditionBaseClass,
+                       public MarkAsCartoon {
+ public:
+  using options = tmpl::list<>;
+  static constexpr Options::String help{
+      "Cartoon boundary condition, to be used in system that do not implement "
```
Comment (nilsdeppe) [inline src/Domain/BoundaryConditions/Cartoon.hpp]: `system` -> `systems` (plural)?

File: tests/Unit/Domain/BoundaryConditions/CMakeLists.txt [hunk 2]
```diff
@@ -4,6 +4,7 @@
 set(LIBRARY "Test_DomainBoundaryConditions")
 
 set(LIBRARY_SOURCES
+  Test_Cartoon.cpp
```
Comment (nilsdeppe) [inline tests/Unit/Domain/BoundaryConditions/CMakeLists.txt]: alphabetize

File: src/Domain/BoundaryConditions/Cartoon.hpp [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+template <typename SystemBoundaryConditionBaseClass>
+// NOLINTNEXTLINE
+PUP::able::PUP_ID Cartoon<SystemBoundaryConditionBaseClass>::my_PUP_ID = 0;
+/// \endcond
+
+/// Check if a boundary condition inherits from `MarkAsCartoon`, which
+/// constitutes as it being marked as an internal Cartoon boundary condition.
+bool is_cartoon(const std::unique_ptr<BoundaryCondition>& boundary_condition);
+
+/// Check if a mesh is compatable with a Cartoon boundary condition, i.e. it is
```
Comment (nilsdeppe) [inline src/Domain/BoundaryConditions/Cartoon.hpp]: `compatible` (typo)

File: tests/Unit/Domain/BoundaryConditions/Test_Cartoon.cpp [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+    CHECK(domain::BoundaryConditions::is_cartoon(cartoon_bc));
+
+    // Test with system that doesn't have cartoon BC
+    auto no_cartoon_bc =
+        domain::BoundaryConditions::make_cartoon_boundary_condition<
+            metavars_without_cartoon>();
+    CHECK(no_cartoon_bc == nullptr);
+  }
+  {
+    INFO("Testing compatability check");
```
Comment (nilsdeppe) [inline tests/Unit/Domain/BoundaryConditions/Test_Cartoon.cpp]: typo `compatibility`

File: tests/Unit/Domain/BoundaryConditions/Test_Cartoon.cpp [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+};
+}  // namespace
+
+SPECTRE_TEST_CASE(
+    "Unit.Domain.BoundaryConditions.Cartoon.TemplateMetafunctions",
+    "[Unit][Domain]") {
+  {
+    INFO("Testing basic metafunctions");
+    // Test inherits_from_mark_as_cartoon trait
+    CHECK(domain::BoundaryConditions::detail::inherits_from_mark_as_cartoon<
```
Comment (nilsdeppe) [inline tests/Unit/Domain/BoundaryConditions/Test_Cartoon.cpp]: I think you can make a lot of the metafunction checks `static_assert`s

File: src/Evolution/DiscontinuousGalerkin/Actions/BoundaryConditionsImpl.hpp [hunk 6]
```diff
@@ -696,11 +697,31 @@ void apply_boundary_conditions_on_all_external_faces(
...
+  // subcell version of mesh does)
+  if constexpr (domain::BoundaryConditions::detail::has_cartoon_bc_v<
+                    tmpl::at<factory_classes,
+                             typename System::boundary_conditions_base>>) {
+    const auto& mesh = db::get<::domain::Tags::Mesh<Dim>>(*box);
+    if (not domain::BoundaryConditions::dg_mesh_is_cartoon_compatible(mesh)) {
+      ERROR(
+          "Cartoon boundary conditions require a mesh that is compatible "
+          "with cartoon methods. Did you put a cartoon BC as an external BC? "
+          "This is not allowed. Replace with another BC");
```
Comment (nilsdeppe) [inline src/Evolution/DiscontinuousGalerkin/Actions/BoundaryConditionsImpl.hpp]: Maybe: "You might have used a Cartoon boundary condition on an external boundary condition. Alternatively and less likely, there is a bug. The mesh is: " << mesh

File: src/Evolution/DiscontinuousGalerkin/Actions/BoundaryConditionsImpl.hpp [hunk 7]
```diff
@@ -696,11 +697,31 @@ void apply_boundary_conditions_on_all_external_faces(
...
+    }
+  }
+
   using derived_boundary_conditions = tmpl::remove_if<
       tmpl::at<factory_classes, typename System::boundary_conditions_base>,
       tmpl::or_<
-          std::is_base_of<domain::BoundaryConditions::MarkAsPeriodic, tmpl::_1>,
-          std::is_base_of<domain::BoundaryConditions::MarkAsNone, tmpl::_1>>>;
+          tmpl::or_<std::is_base_of<domain::BoundaryConditions::MarkAsPeriodic,
```
Comment (nilsdeppe) [inline src/Evolution/DiscontinuousGalerkin/Actions/BoundaryConditionsImpl.hpp]: you don't need to nest the `or_`. `or_` is variadic and takes any number and ors them all.

**Review by nilsdeppe — CHANGES_REQUESTED** [review]

**Review by nilsdeppe — COMMENTED** [review]
LGTM please squash and rebase

**Review by nilsdeppe — APPROVED** [review]
